californium-core/src/main/java/org/eclipse/californium/core/CoapResource.java

```diff
@@ -341,20 +342,35 @@ public void checkObserveRelation(Exchange exchange, Response response) {
 		 * Remember that different paths might lead to this resource.
 		 */
 		
-		ObserveRelation relation = exchange.getRelation();
+		final ObserveRelation relation = exchange.getRelation();
 		if (relation == null || relation.isCanceled()) {
 			return; // because request did not try to establish a relation
 		}
 		if (CoAP.ResponseCode.isSuccess(response.getCode())) {
-			response.getOptions().setObserve(notificationOrderer.getCurrent());
-			
+
 			if (!relation.isEstablished()) {
 				relation.setEstablished();
+				// work-around for missing return value of addObserveRelation
+				// maybe replaced by next major version
```

Comment:
Do you mean you add this observer just in case someone override `addObserveRelation`  (or I don't get your point) ?

Reply:
```
public void addObserveRelation(ObserveRelation relation);
```

It's part of a public interface, and FMPOV, part of the "public API", though implementing a specific `CoapResource` is very basic for californium's usage. I don't see, that something prevents someone from overriding that method (a final would do so, but it's not used). so, yes, adding a boolean return breaks in my opinion the API and therefore I added the work-around with the comment.